.github/workflows/rave.yml

```diff
@@ -50,19 +50,25 @@ jobs:
           # Map each version and if the number is in major.minor syntax, append a .0
           versions=$(jq -c '. | map(.version | capture("(?<major>\\d+)\\.(?<minor>\\d+)(\\.(?<patch>\\d+))?") | .major + "." + .minor + "." + (.patch // "0") )' offers.json)
           echo "$versions" > versions.json
+
       - name: Fetch version matrix from workflow input
         if: ${{ github.event_name == 'workflow_dispatch' }}
         run: | #shell
-          echo '["${{ inputs.version }}"]' > versions.json
+          echo "[\"${VERSION}\"]" > versions.json
+        env:
+          VERSION: ${{ github.event.inputs.version }}
+
       - name: Output versions from versions.json
         id: fetch
         run: | #shell
-          echo "versions=$(cat versions.json)" >> "$GITHUB_OUTPUT"
+          echo versions="$(cat versions.json)" >> "$GITHUB_OUTPUT"
```

Comment:
`echo` has a single string parameter.
https://manpages.debian.org/bookworm/coreutils/echo.1.en.html
Although 'versions' is the variable name. But echo does not know it!!

Reply:
> Echo the STRING(s) to standard output.

Technically it can have more strings, but don't drive the blind horse to the concrete wall! (Hungarian saying, his forehead will be bloody)

Reply:
Yeah this is a leftover habit from attempting to protect against the value defining further variables, for example:

```
123; szepe=viktor
```

However that doesn't work here because the string all gets echoed to `$GITHUB_OUTPUT` anyway.

The way that GHA uses `$GITHUB_OUTPUT` and `$GITHUB_ENV` files is annoying because it's not safe to use with user-defined values.

I need to do some research on how to do this safely ([I'm collecting resources here](https://github.com/stars/johnbillion/lists/github-actions-security)) or avoid it altogether by changing the overall approach.